core/query/evalNestedQuery.ml

```diff
@@ -375,7 +375,10 @@ struct
 
   let rec lins_inner (z, z_fields) ys : QL.t -> QL.t =
     let open QL in
+    let li () = lins_inner (z, z_fields) ys in
```

Comment:
Is there a reason for making li be a function of type `unit -> QL.t -> QL.t` rather than just leaving out the unit?  I guess this avoids nontermination but eta-expanding the RHS by the argument should also do it, and then you'd not have to write `li () xyzzy` everywhere.

```suggestion
    let li e = lins_inner (z, z_fields) ys e in
```